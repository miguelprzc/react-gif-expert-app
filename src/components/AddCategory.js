import React, { useState } from 'react'
import { PropTypes } from 'prop-types';

export const AddCategory = ({setCategories}) => {

  const [inputValue, setInputValue] = useState('');

  const handleInputChange = (event) => {
    setInputValue(event.target.value);
  }

  const handleSubmit = (event) => {
    event.preventDefault();
    if(inputValue.trim()) {
      setCategories(cats => [inputValue, ...cats]);
      setInputValue('');
    }
  }

  return (
    <form onSubmit={handleSubmit}>
      <input 
        type="text"
        placeholder="What are you looking for?"
        value={inputValue}
        onChange={handleInputChange}
      />
    </form>
  )
}

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
}
