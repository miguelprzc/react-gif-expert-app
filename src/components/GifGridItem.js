import React from 'react'
import PropTypes from 'prop-types';

export const GifGridItem = ({title, url}) => {
  return (
    <figure className=" card animate__animated animate__fadeInUp">
      <img src={url} alt={title} />
      <figcaption>{title}</figcaption>
    </figure>
  )
}

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}
