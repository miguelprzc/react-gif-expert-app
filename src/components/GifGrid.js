import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({category}) => {

  const {data: gifs, loading} = useFetchGifs(category);

  return (
    <>
      <h3 className="animate__animated animate__fadeInUp">{category}</h3>
      <div className="card-grid">
        {
          loading && <p className="animate__animated animate__flash">Loading...</p>
        }

        {
          gifs.map((gif) => (
            <GifGridItem 
              key={gif.id} 
              {...gif} 
            />
          ))
        }
      </div>
    </>
  )
}
