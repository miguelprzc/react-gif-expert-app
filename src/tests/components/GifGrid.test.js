import React from 'react';
import '@testing-library/jest-dom'; 

import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('', () => {
  const category = 'Dexter'; 

  test('should render the GifGrid component properly', () => {
     useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    });

    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
  });

  test('should render the GifGrid component properly after the useFecthGifs return a response', () => {
    const gifs = [{id: 'Gif ID', title: 'Gif Title', url: 'Gif URL'},{id: 'Gif ID2', title: 'Gif Title', url: 'Gif URL'}];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    });
    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBeFalsy();
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });
  
  
});