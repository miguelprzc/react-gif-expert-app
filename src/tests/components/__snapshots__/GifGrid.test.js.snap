// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` should render the GifGrid component properly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInUp"
  >
    Dexter
  </h3>
  <div
    className="card-grid"
  >
    <p
      className="animate__animated animate__flash"
    >
      Loading...
    </p>
  </div>
</Fragment>
`;

exports[` should render the GifGrid component properly after the useFecthGifs return a response 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInUp"
  >
    Dexter
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="Gif ID"
      key="Gif ID"
      title="Gif Title"
      url="Gif URL"
    />
    <GifGridItem
      id="Gif ID2"
      key="Gif ID2"
      title="Gif Title"
      url="Gif URL"
    />
  </div>
</Fragment>
`;
